pkg/asset/installconfig/openstack/validation/cloudinfo.go

```diff
@@ -305,6 +317,30 @@ func (ci *CloudInfo) getNetwork(networkName string) (*networks.Network, error) {
 	return network, nil
 }
 
+func (ci *CloudInfo) getNetwork(networkName, networkID string) (*networks.Network, error) {
+	opts := networks.ListOpts{
+		ID:   networkID,
+		Name: networkName,
+	}
+	allPages, err := networks.List(ci.clients.networkClient, opts).AllPages()
+	if err != nil {
+		return nil, err
+	}
+
+	allNetworks, err := networks.ExtractNetworks(allPages)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(allNetworks) == 0 {
+		return nil, nil
+	} else if len(allNetworks) > 1 {
+		return nil, fmt.Errorf("found multiple networks")
+	}
+
+	return &allNetworks[0], nil
+}
+
```

Comment:
Should we check if the subnets extracted above belong to the networks extracted here?